### PR TITLE
Don't emit crossorigin attributes for `output: "export"` by default

### LINKED ISSUE #61210: CORS issues with output: "export" due to unnecessary crossorigin attributes
### Link to the code that reproduces this issue

https://github.com/tdanecker/export-crossorigin-issue

### To Reproduce

```bash
npm run build
```

inspect `./out/index.html`

### Current vs. Expected behavior

Generated link and script tags contain `crossorigin=""`, which is equivalent to `crossorigin="anonymous"`. This forces the browser to do CORS checks, even if the resources could be loaded with no-cors otherwise. See: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin

Example from the generated index.html:
```html
<link rel="stylesheet" href="/_next/static/css/df8ddea784076f40.css" crossorigin="" data-precedence="next"/>
```

**Expected behavior:**

Link and script tags are generated without `crossorigin=""` attributes, as is the case when using other `output` modes.

Example:
```html
<link rel="stylesheet" href="/_next/static/css/df8ddea784076f40.css" data-precedence="next"/>
```

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #40~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 16 10:53:04 UTC 2
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: 3.6.1
  pnpm: 7.33.5
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: export
```

### Which area(s) are affected? (Select all that apply)

Static HTML Export (output: "export")

### Which stage(s) are affected? (Select all that apply)

next build (local), Vercel (Deployed), Other (Deployed)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tdanecker**
In other output modes, link and script tags do not contain a crossorigin attribute by default.

When using output: "export", it should also follow the other output modes, to avoid unnecessary CORS issues, especially when hosting the assets on a different domain (i.e. a CDN).

fixes #61210

### CODE REVIEW

**Comment by hsjobeki** [comment]
Encountering the same issue. Thanks for the fix. Hope this gets merged soon

**Review by styfle — APPROVED** [review]
Thanks for the PR!

It looks like there was a regression introduced when updating types here:

- https://github.com/vercel/next.js/pull/55622

I think this was just a typo so should be safe to land assuming tests pass, thanks!

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 931e2e273627fcd4b138cd3d1556ad3d30f7bf92

`pnpm test test/integration/preload-viewport/test/index.test.js`
- Prefetching Links in viewport > production mode > should prefetch with non-bot UA

<details>
<summary>Expand output</summary>

  ● Prefetching Links in viewport › production mode › should prefetch with non-bot UA

    expect(received).toHaveLength(expected)

    Expected length: 1
    Received length: 0
    Received array:  []

      129 |         )
      130 |         const links = await browser.elementsByCss('link[rel=prefetch]')
    > 131 |         expect(links).toHaveLength(1)
          |                       ^
      132 |       } finally {
      133 |         if (browser) await browser.close()
      134 |       }

      at Object.toHaveLength (integration/preload-viewport/test/index.test.js:131:23)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| buildDuration | 13.7s | 14s | ⚠️ +292ms |
| buildDurationCached | 7.4s | 6.3s | N/A |
| nodeModulesSize | 198 MB | 198 MB | N/A |
| nextStartRea..uration (ms) | 435ms | 434ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 30.2 kB | 30.2 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | N/A |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 242 B | 242 B | ✓ |
| main-HASH.js gzip | 32.1 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 45.6 kB | 45.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.2 kB | 4.2 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.56 kB | 6.56 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 485 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| index.html gzip | 530 B | 529 B | N/A |
| link.html gzip | 542 B | 542 B | ✓ |
| withRouter.html gzip | 524 B | 523 B | N/A |
| Overall change | 542 B | 542 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95 kB | 95 kB | N/A |
| page.js gzip | 3.06 kB | 3.07 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 625 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 990 B | 990 B | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 171 kB | 171 kB | ✓ |
| app-page-exp..prod.js gzip | 96.7 kB | 96.7 kB | ✓ |
| app-page-tur..prod.js gzip | 98.5 kB | 98.5 kB | ✓ |
| app-page-tur..prod.js gzip | 92.9 kB | 92.9 kB | ✓ |
| app-page.run...dev.js gzip | 150 kB | 150 kB | ✓ |
| app-page.run..prod.js gzip | 91.4 kB | 91.4 kB | ✓ |
| app-route-ex...dev.js gzip | 21.3 kB | 21.3 kB | ✓ |
| app-route-ex..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 14.8 kB | 14.8 kB | ✓ |
| app-route.ru...dev.js gzip | 20.9 kB | 20.9 kB | ✓ |
| app-route.ru..prod.js gzip | 14.7 kB | 14.7 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.51 kB | 9.51 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.79 kB | 9.79 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.51 kB | 9.51 kB | ✓ |
| pages-turbo...prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| pages.runtim...dev.js gzip | 23 kB | 23 kB | ✓ |
| pages.runtim..prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| server.runti..prod.js gzip | 50.6 kB | 50.6 kB | ✓ |
| Overall change | 950 kB | 950 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tdanecker/next.js bug/export-crossorigin | Change |
| - | - | - | - |
| 0.pack gzip | 1.56 MB | 1.55 MB | N/A |
| index.pack gzip | 105 kB | 106 kB | ⚠️ +451 B |
| Overall change | 105 kB | 106 kB | ⚠️ +451 B |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 931e2e273627fcd4b138cd3d1556ad3d30f7bf92

File: packages/next/src/export/index.ts [hunk 1]
```diff
@@ -399,7 +399,7 @@ export async function exportAppImpl(
     disableOptimizedLoading: nextConfig.experimental.disableOptimizedLoading,
     // Exported pages do not currently support dynamic HTML.
     supportsDynamicHTML: false,
-    crossOrigin: nextConfig.crossOrigin || '',
+    crossOrigin: nextConfig.crossOrigin || undefined,
```
Comment (styfle) [inline packages/next/src/export/index.ts:402]:
```suggestion
    crossOrigin: nextConfig.crossOrigin,
```

I'm not sure if this is intended but your code was coercing the empty string to undefined. Perhaps we should leave it as is?

**Pull request merged**